frontend/packages/console-app/src/components/quick-starts/catalog/QuickStartCatalog.tsx

```diff
@@ -38,30 +49,90 @@ const QuickStartCatalog: React.FC<QuickStartCatalogProps> = ({
   setActiveQuickStart,
 }) => {
   const { t } = useTranslation();
+
+  const [filterText, setFilterText] = React.useState('');
+  const [filterStatus, setFilterStatus] = React.useState([]);
+  const onFilter = (val: string, statusFilters: QuickStartFilteredStatus) => {
```

Comment:
Use `useCallback`. Probably better to split this into two `handleTextFilterChange` and `handleStatusFilterChange`.

Reply:
Updated, split into two functions. IMO `handleTextFilterChange` is very simple and doesn't need to be turned into a callback. I've added `useCallback` for `handleStatusFilterChange`.